scripts/esbuild/sys-node.ts

```diff
@@ -36,10 +36,15 @@ export async function buildSysNode(opts: BuildOptions) {
     // normally you wouldn't externalize your "own" directory here, but since
     // we build multiple things within `opts.output.sysNodeDir` which should
     // externalize each other we need to do so
-    join(opts.output.sysNodeDir, '*'),
+    '../../compiler/stencil.js',
+    '../../sys/node/*',
   ];
 
-  const sysNodeAliases = getEsbuildAliases();
+  const sysNodeAliases = {
+    ...getEsbuildAliases(),
+    '@stencil/core/compiler': '../../compiler/stencil.js',
```

Comment:
If this path is different than what's spit out by `getEsbuildAliases()`, does it make sense to have that function generate the path based on some directory? Why are they different in the first place?

Reply:
there are some build artifacts that located multiple folders deep that require a different alias path, e.g. for `/testing/index.js` the alias for `@stencil/core/compiler` has to be `../compiler/stencil.js` while for `/sys/node/index.js` it has to be `../../compiler/stencil.js`.

Reply:
Right, so I was thinking we could figure out how deeply nested we are based on the relative path from `__filename` to `src` or something and use that info to generate the relative paths.

Or, alternatively, remove the helper altogether. What you have works, but feels a bit odd that we use it and then override the returned aliases in quite a few spots.

Open to thoughts!

Reply:
Most Stencil components can be successfully compiled using the common defined aliases as implemented in [scripts/esbuild/utils/index.ts](https://github.com/ionic-team/stencil/pull/5930/files#diff-ca5c0dbb4baece2d3fbe79900e692152217277778ef3679274a74c5c0ee4057d), there are only two artifacts that require modifications to these aliases which is:

- one artifact in `scripts/esbuild/sys-node.ts`
- and `scripts/esbuild/internal-platform-testing.ts`

If you want I can get rid of the `getEsbuildAliases` function and have this defined per component but I think this will create more code than we have right now.

Wdyt?

Reply:
I think it's fine for now if it's only a couple spots. Thanks for the pushback, no need to overcomplicate